File: web-app/online-warehouse/src/app/commodity-lot/dto/create-commodity-lot.dto.ts
import {CommodityLotTypeEnum} from "./enum/commodity-lot-type.enum";
import {CreateCommodityLotGoodsDto} from "./create-commodity-lot-goods.dto";

export class CreateCommodityLotDto {
  counterpartyId: number;
  commodityLotType: CommodityLotTypeEnum;
  createCommodityLotGoodsDtoList: Array<CreateCommodityLotGoodsDto>;


  constructor(counterpartyId: number, commodityLotType: CommodityLotTypeEnum, createCommodityLotGoodsDtoList: Array<CreateCommodityLotGoodsDto>) {
    this.counterpartyId = counterpartyId;
    this.commodityLotType = commodityLotType;
    this.createCommodityLotGoodsDtoList = createCommodityLotGoodsDtoList;
  }
}

File: core/src/main/java/by/itechart/commoditylot/dto/CreateCommodityLotDto.java
package by.itechart.commoditylot.dto;

import by.itechart.commoditylot.enums.CommodityLotType;
import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.util.List;

@Data
public class CreateCommodityLotDto {
    private Long counterpartyId;
    @NotNull
    private CommodityLotType commodityLotType;
    @NotEmpty
    private List<CreateCommodityLotGoodsDto> createCommodityLotGoodsDtoList;
}
